connector/spanmetricsconnectorv2/config/config.go

```diff
@@ -57,6 +60,82 @@ type Config struct {
 	Spans []MetricInfo `mapstructure:"spans"`
 }
 
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+func (c *Config) Validate() error {
+	duplicate := make(map[string]MetricInfo)
+	for _, info := range c.Spans {
+		if old, ok := duplicate[info.Name]; ok && info.isEqual(old) {
+			return fmt.Errorf("spans: duplicate configuration found %s", info.Name)
+		}
+		if info.Name == "" {
+			return errors.New("spans: metric name missing")
+		}
+		if info.Unit == "" {
+			return errors.New("spans: metric unit missing")
+		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spans attributes validation failed: metric %q: %w", info.Name, err)
+		}
+		if info.noAggregatorDefined() {
+			return errors.New("metric definition missing, either histogram, summary, or counters required")
+		}
+		if err := info.validateHistogram(); err != nil {
+			return fmt.Errorf("spans histogram validation failed: metric %q, %w", info.Name, err)
+		}
+		duplicate[info.Name] = info
+	}
+	return nil
+}
+
+// Unmarshal with custom logic to set default values.
+// This is necessary to ensure that default metrics are
+// not configured if the user has specified any custom metrics.
+func (c *Config) Unmarshal(componentParser *confmap.Conf) error {
+	if componentParser == nil {
+		// Nothing to do if there is no config given.
+		return nil
+	}
+	if err := componentParser.Unmarshal(c, confmap.WithIgnoreUnused()); err != nil {
+		return err
+	}
+	if !componentParser.IsSet("spans") {
+		c.Spans = defaultSpansConfig()
+		return nil
+	}
+	for k, info := range c.Spans {
+		if info.Unit == "" {
+			info.Unit = MetricUnitMs
+		}
+		if info.noAggregatorDefined() {
+			info.Histogram.Exponential = &ExponentialHistogram{
+				MaxSize: defaultExponentialHistogramMaxSize,
+			}
+		}
+		if info.Histogram.Explicit != nil {
+			// Add default buckets if explicit histogram is defined
+			if len(info.Histogram.Explicit.Buckets) == 0 {
+				info.Histogram.Explicit.Buckets = defaultHistogramBuckets[:]
+			}
+		}
+		if info.Histogram.Exponential != nil {
+			if info.Histogram.Exponential.MaxSize == 0 {
+				info.Histogram.Exponential.MaxSize = defaultExponentialHistogramMaxSize
+			}
+		}
+		if info.Counters != nil {
+			if info.Counters.SumSuffix == "" {
+				info.Counters.SumSuffix = defaultCountersSumSuffix
+			}
+			if info.Counters.CountSuffix == "" {
+				info.Counters.CountSuffix = defaultCountersCountSuffix
+			}
+		}
+		c.Spans[k] = info
+	}
+	return nil
+}
+
```

Comment:
[**For reviewers**] This bit of code is mostly moved from below. The only change is using `info.noAggregatorDefined` to check if no aggregator has been defined for a metric configuration.